pkg/gds/models/v1/models.go

```diff
@@ -225,6 +225,35 @@ func AppendCertID(vasp *pb.VASP, certID string) (err error) {
 	return nil
 }
 
+// NewCertificate creates and returns a certificate associated with a VASP. This will
+// error if the VASP does not contain an identity certificate.
+func NewCertificate(vasp *pb.VASP, certRequest *CertificateRequest) (cert *Certificate, err error) {
+	// VASP must be not nil.
+	if vasp == nil {
+		return nil, errors.New("must supply a VASP object for certificate creation")
+	}
+
+	// Certificate request must be not nil.
+	if certRequest == nil {
+		return nil, errors.New("must supply a certificate request for certificate creation")
+	}
+
+	// VASP must contain an identity certificate.
+	if vasp.IdentityCertificate == nil {
+		return nil, errors.New("VASP must contain an identity certificate")
+	}
+
+	cert = &Certificate{
+		Id:      uuid.New().String(),
```

Comment:
The way that we identify the certificate in Sectigo is via the capital hex-encoded serial number; may I propose that we use that serial number here instead of a random UUID? Or if you think we should be more specific, could we include a Serial Number field at the top level that ensures the certificate serial number is rendered in such a way that it can be looked up using the Sectio UI?

Reply:
I pushed some changes to use the serial number instead of a random ID, can you check to make sure I interpreted your suggestions correctly?